pages/CitationStyleLanguageHandler.inc.php

```diff
@@ -79,10 +79,19 @@ public function download($args, $request)
         $this->_setupRequest($args, $request);
 
         $plugin = PluginRegistry::getPlugin('generic', 'citationstylelanguageplugin');
-        $plugin->downloadCitation($request, $this->article, $this->citationStyle, $this->issue, $this->publication);
+        $plugin->downloadCitation($request, $this->submission, $this->citationStyle, $this->issue);
         exit;
     }
 
+    protected function isSubmissionUnpublished($issue, $submission) {
+        $applicationName = Application::get()->getName();
+
+        if($applicationName == 'ojs2')
+            return !$issue || !$issue->getPublished() || $submission->getStatus() != PKPSubmission::STATUS_PUBLISHED;
+        else if ($applicationName == 'ops')
+            return $submission->getStatus() != PKPSubmission::STATUS_PUBLISHED;
```

Comment:
Add brackets to these conditionals:

```php
if (...) {
    return ...;
} else if {
    return ...;
}
```

Reply:
Also, let's make the last condition the default. This should also work for OMP, which this plugin will eventually support:

```php
if ($applicationName === 'ojs2') {
    return ...;
}

return ...;
```